### PR TITLE
Port for Intel Aero

### PR DESCRIPTION
**Pull request by guludo**
Hi everyone!

This adds support for running Ardupilot on the Intel Aero board :-)

Best regards,
Gustavo Sousa

### CODE REVIEW

File: libraries/AP_Compass/AP_Compass_BMM150.cpp [hunk 1]
```diff
@@ -239,25 +241,20 @@ int16_t AP_Compass_BMM150::_compensate_z(int16_t z, uint32_t rhall)
     return constrain_int32(dividend / divisor, -0x8000, 0x8000);
 }
 
-void AP_Compass_BMM150::_update()
```
Comment (magicrub) [inline libraries/AP_Compass/AP_Compass_BMM150.cpp]: would be nice to have a comment explaining what returning true/false means

File: libraries/AP_Compass/AP_Compass_BMM150.cpp [hunk 2]
```diff
@@ -283,28 +280,34 @@ void AP_Compass_BMM150::_update()
...
     _mag_accum += raw_field;
     _accum_count++;
     if (_accum_count == 10) {
         _mag_accum /= 2;
         _accum_count = 5;
     }
-
     _last_update_timestamp = time_usec;
+    _accum_sem->give();
 }
```
Comment (magicrub) [inline libraries/AP_Compass/AP_Compass_BMM150.cpp]: does not explicitly return a bool

File: libraries/AP_HAL/AP_HAL_Boards.h [hunk 3]
```diff
@@ -70,6 +71,7 @@
 #define HAL_INS_QFLIGHT  15
 #define HAL_INS_QURT     16
 #define HAL_INS_BBBMINI  17
+#define HAL_INS_AERO 18
```
Comment (magicrub) [inline libraries/AP_HAL/AP_HAL_Boards.h]: tab needed

File: libraries/AP_HAL_Linux/HAL_Linux_Class.cpp [hunk 4]
```diff
@@ -148,6 +150,8 @@ static RCInput_DSM rcinDriver;
 static RCInput_SBUS rcinDriver;
 #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_NAVIO2
 static RCInput_Navio2 rcinDriver;
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_AERO
```
Comment (magicrub) [inline libraries/AP_HAL_Linux/HAL_Linux_Class.cpp]: combine with HAL_BOARD_SUBTYPE_LINUX_DISCO just like the NAVIO, ERLE, BH, PXFMINI one does?

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 5]
```diff
@@ -567,6 +567,14 @@ AP_InertialSensor::detect_backends(void)
     } else {
         hal.console->printf("MPU9250: External IMU not detected\n");
     }
+#elif HAL_INS_DEFAULT == HAL_INS_AERO
+    auto *backend = AP_InertialSensor_BMI160::probe(*this,
+                                                    hal.spi->get_device("bmi160"));
+    if (backend) {
+        _add_backend(backend);
+    } else {
+        hal.console->printf("aero: onboard IMU not detected\n");
```
Comment (magicrub) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp]: should a panic go here? If there's no IMU then that's a catastrophic failure

Comment (magicrub) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp]: ahh, I see it happens below via if (_backend_count == 0). nevermind.

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI160.cpp [hunk 6]
```diff
@@ -407,15 +414,10 @@ void AP_InertialSensor_BMI160::_read_fifo()
     }
 }
 
-void AP_InertialSensor_BMI160::_poll_data()
+bool AP_InertialSensor_BMI160::_poll_data()
 {
-    if (!_dev->get_semaphore()->take_nonblocking()) {
```
Comment (magicrub) [inline libraries/AP_InertialSensor/AP_InertialSensor_BMI160.cpp]: does removing this break other hardware that use this driver?

File: libraries/AP_Compass/AP_Compass_BMM150.cpp (hunk 1 above)
Comment (guludo) [inline libraries/AP_Compass/AP_Compass_BMM150.cpp]: There's a comment on the header.

File: libraries/AP_Compass/AP_Compass_BMM150.cpp (hunk 2 above)
Comment (guludo) [inline libraries/AP_Compass/AP_Compass_BMM150.cpp]: Ooops! Thanks!

File: libraries/AP_HAL/AP_HAL_Boards.h (hunk 3 above)
Comment (guludo) [inline libraries/AP_HAL/AP_HAL_Boards.h]: Ok

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI160.cpp (hunk 6 above)
Comment (lucasdemarchi) [inline libraries/AP_InertialSensor/AP_InertialSensor_BMI160.cpp]: the lock is now implicit in the bus... and no other board uses this driver... we added it a while ago exactly for this board.

**Comment by tridge** [comment]
this looks great! please push when ready

**Comment by magicrub** [comment]
this would be nice to have in the Plane 3.7 release :)

File: libraries/AP_Compass/AP_Compass_BMM150.cpp (hunk 2 above)
Comment (muramura) [inline libraries/AP_Compass/AP_Compass_BMM150.cpp]: this method end, not return value.
return vlue is invalid.

File: libraries/AP_HAL_Linux/HAL_Linux_Class.cpp (hunk 4 above)
Comment (guludo) [inline libraries/AP_HAL_Linux/HAL_Linux_Class.cpp]: ack

File: libraries/AP_HAL/AP_HAL_Boards.h (hunk 3 above)
Comment (guludo) [inline libraries/AP_HAL/AP_HAL_Boards.h]: Hm, now I see why I did it like that, the other ones aren't either hehe

This is just my opinion, but I don't think it's worth trying to keep things like that aligned. When a new macro with a bigger name is added we need to align everyone again. If that is done in the same commit, then it doesn't highlight to the reader what's been added; if in a separate commit, then we have another patch.

Anyway, I've added a patch reformatting these and other defines, will push soon with the other fixes.

**Comment by guludo** [comment]
Force pushed with the fixes.

File: libraries/AP_HAL_Linux/RCOutput_AeroIO.cpp [hunk 7]
```diff
@@ -15,20 +15,12 @@
...
 
-#include <dirent.h>
-#include <fcntl.h>
-#include <stdint.h>
-#include <stdio.h>
-#include <stdlib.h>
 #include <utility>
-#include <sys/stat.h>
-#include <sys/types.h>
-#include <unistd.h>
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/RCOutput_AeroIO.cpp]: humn... you forgot to squash this commit.

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 8]
```diff
@@ -567,6 +567,14 @@ AP_InertialSensor::detect_backends(void)
     } else {
         hal.console->printf("MPU9250: External IMU not detected\n");
     }
```
Comment (lucasdemarchi) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp]: in the commit message: BMI160, not BMP160.

**Comment by lucasdemarchi** [comment]
Applied with the above fixes.

**Comment by rmackay9** [comment]
Great!  I'll include this with -rc4 so Aero support will go out with AC3.4.

**Pull request closed without merging**